strawberryfields/plot.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+This module provides tools to visualize the state in various interactive
+ways using Plot.ly
+"""
+from copy import deepcopy
+
+import numpy as np
+import plotly.io as pio
+
+
+def plot_wigner(state, renderer="browser", wire=0, dq=0.1):
+    """Plot the Wigner function with Plot.ly
+
+    Args:
+        state (:class:`.BaseState`): the state used for plotting
+        renderer (string): the renderer for plotting with Plot.ly
+        wire (int): wire used to calculate the reduced Wigner function
+        dq (float): stepsize along the q and p axes
+    """
+    q = np.arange(-state.cutoff_dim, state.cutoff_dim, dq)
```

Comment:
Will `cutoff_dim` be defined for Gaussian states?

We should probably allow the user to pass their _own_ value of the p and q axis to this function, for full control. E.g.,

```python
def plot_wigner(state, p_axis, q_axis, wires=0, renderer="browser")
```
(or something like that)

Reply:
> Will cutoff_dim be defined for Gaussian states?

No, this doesn't work with Gaussian states the way I wrote it. This was just a way SF interactive did it. I was planning to update this, but hadn't decided on exactly how to do it yet. Either just as you suggested or having the option to just call `plot_wigner(state)` for the Fock backend, which would use the cutoff (and raise an "must-define-p-and-q" error for the Gaussian), and otherwise having to write e.g. like your suggestion.